### PR TITLE
V1 refactor

### PR DESCRIPTION
**Pull request by tbody-cfs**
# Major refactor
1. Simplifying and tidying up the interface.
2. Now uses the same grid as the OpenADAS files, which avoids an off-grid extrapolation warning.
3. More robust testing
4. `pip install -i https://test.pypi.org/simple/ radas` now works
5. Code has been significantly tidied up
6. Out of source builds, makes it easier to install once as an admin and then have the code available for users

### CODE REVIEW

**Comment by tbody-cfs** [comment]
@mikekryjak and @bshanahan — if you get a chance, could you let me know if this improves usability of radas?

**Review by hassec — CHANGES_REQUESTED** [review]
Some quick first comments. 

Haven't had the time yet to really figure out what was moved and what is new to do a full review.

Is the goal to also turn this into a PyPi package? 
Have you checked if the `cfspopcon` pipelines will still work after these changes?

File: pyproject.toml [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
+Pint = "^0.23"
 pyyaml = "^6.0"
-xarray = "^2023.3.0"
+xarray = "^2024.2.0"
 pint-xarray = "^0.3"
-click = "^8.1.3"
+click = "^8.1.7"
+black = "^24.2.0"
+pytest-order = "^1.2.0"
+ipdb = "^0.13.13"
```
[the remark this answers is not in the document]
Comment (tbody-cfs) [reply pyproject.toml]: Moved to `dev` group

File: radas/adas_interface/determine_adas_dataset_type.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+def determine_reader_class_and_config(data_file_config, dataset_type):
+    """Examines the data_file_config to determine which reader class to use to reader a specific dataset_type."""
```
[the remark this answers is not in the document]
Comment (tbody-cfs) [reply radas/adas_interface/determine_adas_dataset_type.py]: Fixed

File: tests/test_against_mavrin.py [hunk 3]
```diff
@@ -0,0 +1,2 @@
+def compare_against_mavrin():
+    pass
```
[the remark this answers is not in the document]
Comment (tbody-cfs) [reply tests/test_against_mavrin.py]: Deleted, now in `test_main.py`

File: radas/shared.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
+"""Bits and pieces shared by many other parts of radas."""
+
+from pathlib import Path
+import subprocess
+import yaml
+
+module_directory = Path(__file__).parent
+data_file_directory = module_directory / ".data_files"
+repository_directory = module_directory.parent
+output_directory = repository_directory / "output"
```
[the remark this answers is not in the document]
Comment (tbody-cfs) [reply radas/shared.py]: Now using `importlib-resources`

File: pyproject.toml [hunk 5]
```diff
@@ -1,31 +1,49 @@
...
+Pint = "^0.21"
 pyyaml = "^6.0"
-xarray = "^2023.3.0"
-pint-xarray = "^0.3"
-click = "^8.1.3"
-
+xarray = ">=2023"
+pint-xarray = ">=0.3"
+click = "^8.1"
+importlib-resources = ">=5.10"
```
Comment (hassec) [inline pyproject.toml:36]: this is only required for older python version. [`importlib.resources`](https://docs.python.org/3.9/library/importlib.html?highlight=importlib#module-importlib.resources) is part of the standard library

Comment (tbody-cfs) [reply pyproject.toml]: Removed this now

**Pull request merged**